### PR TITLE
Fix malform json

### LINKED ISSUE #188: Application does not launch when JSON is malformed
This is when fields that are supposed to be of type `string` are replaced by the value `null`.

image

### LINKED ISSUE #212: [PE-D][Tester A] Parameter for email
Should the domain for email contain a single . ? Just like the domain for links.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: Blizzeracz/ped#4

### PR DESCRIPTION
**Pull request by nknguyenhc**
* If any value is `null`, the programme resorts to default values.
* If there are duplicates in a unique list, the programme resorts to an empty model.
* Email regex is fixed to fit the description in the user guide.

### CODE REVIEW

File: src/main/java/networkbook/MainApp.java [hunk 1]
```diff
@@ -88,6 +89,10 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             logger.warning("Data file at " + storage.getNetworkBookFilePath() + " could not be loaded."
                     + " Will be starting with an empty NetworkBook.");
             initialData = new NetworkBook();
+        } catch (NullValueException e) {
+            logger.warning(e.getMessage());
+            logger.warning("Starting with an empty NetworkBook.");
+            initialData = new NetworkBook();
```
Comment (Singa-pirate) [inline src/main/java/networkbook/MainApp.java:95]: I feel we can set the output text of the command box to notify user that (1) data file is invalid, (2) reason why it's invalid (basically `e.getMessage()`), (3) app is initialized with empty data which will overwrite original data file

**Review by Singa-pirate — COMMENTED** [review]
LGTM in general except for a small suggestion to initialise the command box output text if json is malformed

File: src/main/java/networkbook/MainApp.java (hunk 1 above)
Comment (nknguyenhc) [reply src/main/java/networkbook/MainApp.java:95]: This is feature change though.

Comment (Singa-pirate) [reply src/main/java/networkbook/MainApp.java:95]: Right, KIV for future development.

**Review by Singa-pirate — APPROVED** [review]

**Pull request merged**